pkg/occlient/occlient_test.go

```diff
@@ -15,6 +15,100 @@ import (
 	ktesting "k8s.io/client-go/testing"
 )
 
+func TestAddPVCToDeploymentConfig(t *testing.T) {
+	type args struct {
+		dc   *appsv1.DeploymentConfig
+		pvc  string
+		path string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Test case : valid dc",
+			args: args{
+				dc: &appsv1.DeploymentConfig{
+					Spec: appsv1.DeploymentConfigSpec{
+						Selector: map[string]string{
+							"deploymentconfig": "nodejs-app",
+						},
+						Template: &corev1.PodTemplateSpec{
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{
+									{
+										Name: "test",
+										VolumeMounts: []corev1.VolumeMount{
+											{
+												MountPath: "/tmp",
+												Name:      "test",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				pvc:  "testglocks volume",
+				path: "/mnt",
+			},
+			wantErr: false,
+		},
+		{
+			name: "Test case: dc without Containers defined",
+			args: args{
+				dc: &appsv1.DeploymentConfig{
+					Spec: appsv1.DeploymentConfigSpec{
+						Selector: map[string]string{
+							"deploymentconfig": "nodejs-app",
+						},
+						Template: &corev1.PodTemplateSpec{
+							Spec: corev1.PodSpec{},
+						},
+					},
+				},
+				pvc:  "my xx voulme",
+				path: "/mnt",
+			},
+			wantErr: true,
+		},
+		{
+			name: "Test case: dc without Template defined",
+			args: args{
+				dc: &appsv1.DeploymentConfig{
+					Spec: appsv1.DeploymentConfigSpec{
+						Selector: map[string]string{
+							"deploymentconfig": "nodejs-app",
+						},
+					},
+				},
+				pvc:  "my xx voulme",
+				path: "/mnt",
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient, fakeClientset := FakeNew()
+
+			fakeClientset.AppClientset.PrependReactor("update", "deploymentconfigs", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, nil, nil
```

Comment:
We should return the updated dc instead of nil and verify that the dc was updated with the proper values as done in https://github.com/redhat-developer/odo/blob/e8e223b6edd6f046035ec1ad249448303fbf4b67/pkg/occlient/occlient_test.go#L1337

Also we should check that update was called on the correct dc by verifying against the name of the dc
https://github.com/redhat-developer/odo/blob/e8e223b6edd6f046035ec1ad249448303fbf4b67/pkg/occlient/occlient_test.go#L1323-L1326

Reply:
I agree that it is good to validate the values with which the function is being called but in this specific case there is a random string generation involved here https://github.com/redhat-developer/odo/blob/master/pkg/occlient/occlient.go#L1333
so it won't be feasible.
will add validation for the number of action performed

Reply:
added validation for the presence of mounts and with passing volume names & rebased

now ready for review